rest/src/main/java/io/syndesis/rest/v1/handler/setup/OAuthAppHandler.java

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.rest.v1.handler.setup;
+
+import io.swagger.annotations.Api;
+import io.syndesis.core.SuppressFBWarnings;
+import io.syndesis.dao.manager.DataManager;
+import io.syndesis.model.connection.ConfigurationProperty;
+import io.syndesis.model.connection.Connector;
+import org.springframework.stereotype.Component;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeSet;
+
+/**
+ * This rest endpoint handles working with global oauth settings.
+ */
+@Path("/setup/oauth-apps")
+@Api(value = "oauth-apps")
+@Component
+public class OAuthAppHandler {
+
+    private final DataManager dataMgr;
+
+    public OAuthAppHandler(DataManager dataMgr) {
+        this.dataMgr = dataMgr;
+    }
+
+    // Since this a a view model DTO, and not a domain model lets define it here instead
+    // of placing it into the model module.
+    @SuppressFBWarnings(
+        value = "URF_UNREAD_PUBLIC_OR_PROTECTED_FIELD",
+        justification = "All fields are encode by jackson and sent to the UI")
+    public static final class OAuthApp {
```

Comment:
Would it make sense to use Immutables nonetheless and have a `from(Connector)` method?